r/R/dplyr-funcs.R

```diff
@@ -160,6 +160,17 @@ create_binding_cache <- function() {
   register_bindings_type()
   register_bindings_augmented()
 
+  # HACK because random() doesn't work (ARROW-17974)
+  register_scalar_function(
+    "_random_along",
+    function(context, x) {
+      Array$create(runif(length(x)))
+    },
```

Comment:
Cool, thanks, I'll experiment. Weston linked to some other jiras that suggest that a function with 0 args won't work for other reasons, so that may not end up working, but we'll see.

Reply:
I tried this, and it fails with the same error message as the C++ `random()` function: `Error in `collect(.)`: Invalid: ExecuteScalarExpression cannot Execute non-scalar expression (Array[double] < 0.2)`

That suggests that the problem with `random()` isn't random per se but the other issues that Weston flagged.

(FTR, the parameter is `context$batch_length`)